### PR TITLE
Fix #271: Make region field optional.

### LINKED ISSUE #271: Job Board: "Region" should not be a required field
In some cases (particularly US states), this field has value, but in others a job submission will have just a City and Country. This should be allowed, rather than forcing the submitter to put something in this field.

This might require a database migration; I'm not sure.

### PR DESCRIPTION
**Pull request by SarahPythonista**

### CODE REVIEW

File: jobs/feeds.py [hunk 1]
```diff
@@ -18,7 +18,8 @@ def item_title(self, item):
 
     def item_description(self, item):
         """ Description """
-        location = ",".join([item.city, item.region, item.country])
+        location = ",".join(location_part for location_part in\
```
Comment (berkerpeksag) [inline jobs/feeds.py]: no need to add `\` here. I'd write this something like

``` py
location_parts = (item.city, item.region, item.country)
locaiton = ",".join(location_part for location_part in location_parts
                    if location_part is not None)
```

File: jobs/models.py [hunk 2]
```diff
@@ -100,7 +100,11 @@ def __str__(self):
         return 'Job Listing #{}'.format(self.pk)
 
     def save(self, **kwargs):
-        self.location_slug = slugify('%s %s %s' % (self.city, self.region, self.country))
+        location_str = ''
+        for location_part in (self.city, self.region, self.country):
+            if location_part is not None:
+                location_str = ' '.join(location_str, location_part)
```
Comment (berkerpeksag) [inline jobs/models.py]: same as above. also, `' '.join(location_str, location_part)` should be `' '.join([location_str, location_part])`

Thanks!

**Comment by berkerpeksag** [comment]
Could you add a test to https://github.com/python/pythondotorg/blob/master/jobs/tests/test_models.py?

**Comment by SarahPythonista** [comment]
Fixed the issues with the original commit. Added a test. 

Thanks @berkerpeksag

**Comment by SarahPythonista** [comment]
I discovered while working on #579 that some of the templates also need to be edited.

I'll get to that tomorrow.

**Comment by frankwiles** [comment]
Merged.  Thanks!

**Pull request merged**